### PR TITLE
Deprecated integrations playbooks scripts hook

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/23417

## Description
fix all deprecated files to shape of deprecated

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/tests/integration_tests/validate_integration_test.py [hunk 1]
```diff
@@ -1052,6 +1113,45 @@ def test_invalid_playbook(self, mocker):
         assert result.exit_code == 1
 
 
+class TestPlaybookValidateDeprecated:
+    def test_valid_playbook_deprecated(self, mocker, repo):
```
Comment (guykeller) [inline demisto_sdk/tests/integration_tests/validate_integration_test.py:1117]: maybe test_valid_deprecated playbook?
same order with other functions

File: demisto_sdk/tests/integration_tests/validate_integration_test.py [hunk 2]
```diff
@@ -1194,6 +1294,58 @@ def test_invalid_script(self, mocker, repo):
...
+        Given
+        - a valid deprecated Script.
+
+        When
+        - Running validate on it.
+
+        Then
+        - Ensure validate passes and identifies the file as a deprecated script.
+        """
+        mocker.patch.object(tools, 'is_external_repository', return_value=True)
```
Comment (guykeller) [inline demisto_sdk/tests/integration_tests/validate_integration_test.py:1309]: why is this needed?

File: demisto_sdk/tests/integration_tests/validate_integration_test.py [hunk 3]
```diff
@@ -91,6 +95,63 @@ def test_invalid_incident_field(self, mocker, repo):
         assert "The system key must be set to False" in result.stdout
 
 
+class TestDeprecatedValidation:
```
Comment (guykeller) [inline demisto_sdk/tests/integration_tests/validate_integration_test.py:98]: did you mean `TestDeprecatedIntegration`?

File: demisto_sdk/tests/test_files/Packs/CortexXDR/Playbooks/Invalid_Deprecated_Playbook.yml [hunk 4]
```diff
@@ -0,0 +1,644 @@
+id: Cortex XDR Incident Handling
+version: -1
+name: Cortex XDR Incident Handling
+description: "This playbook is triggered by fetching a Palo Alto Networks Cortex XDR
```
Comment (guykeller) [inline demisto_sdk/tests/test_files/Packs/CortexXDR/Playbooks/Invalid_Deprecated_Playbook.yml:4]: Do we really need to use test files, or can we use the test utils that @jochman created?
Same for the valid file.

File: demisto_sdk/commands/common/hook_validations/base_validator.py [hunk 5]
```diff
@@ -98,8 +98,7 @@ def check_deprecated(self, file_path):
         if file_path.endswith('.yml'):
             yml_dict = get_yaml(file_path)
             if ('deprecated' in yml_dict and yml_dict['deprecated'] is True) or \
-                    (file_name.startswith('playbook') and 'hidden' in yml_dict and
-                     yml_dict['hidden'] is True):
+                    (file_name.startswith('playbook') and 'hidden' in yml_dict and yml_dict['hidden'] is True):
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/base_validator.py:101]: In a conversation with Roy he told me to change here something he had written before because it did not look good, and after I changed there were tests that did not pass, so I changed it back.

File: demisto_sdk/tests/test_files/Packs/CortexXDR/Playbooks/Invalid_Deprecated_Playbook.yml (hunk 4 above)
Comment (ChanochShayner) [reply demisto_sdk/tests/test_files/Packs/CortexXDR/Playbooks/Invalid_Deprecated_Playbook.yml:4]: I tried to use a playbook that already exists for testing and failed, so I created a new file.

File: demisto_sdk/commands/common/hook_validations/base_validator.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/common/hook_validations/base_validator.py:101]: In a conversation with @roysagi  he told me to change here something he had written before because it did not look good, and after I changed there were tests that did not pass, so I changed it back.

[the remark this answers is not in the document]
Comment (roysagi) [reply demisto_sdk/commands/common/hook_validations/base_validator.py:101]: @ChanochShayner , @guykeller - As I told Chanoch the file name does not have to start with "playbook" to be a playbook, which makes this code incorrect - therefore in need of a fix.
@ChanochShayner - please revert to the changes we made and check which unit tests failed - we should fix the unit tests and not ignore this change.

File: demisto_sdk/commands/common/errors.py [hunk 6]
```diff
@@ -367,12 +370,27 @@ def invalid_v2_integration_name():
     def found_hidden_param(parameter_name):
         return f"Parameter: \"{parameter_name}\" can't be hidden. Please remove this field."
 
+    @staticmethod
+    @error_code_decorator
+    def invalid_deprecated_integration(display_name, description):
+        error_answer = ''
+        if not display_name.startswith('Deprecated.'):
+            error_answer += 'The display_name (display) of your integration should start with "Deprecated.".\n'
```
Comment (guykeller) [inline demisto_sdk/commands/common/errors.py:378]: Let's make this change everywhere, to make it clear why the word deprecate should appear
```suggestion
            error_answer += 'The display_name (display) of all deprecated integrations should start with "Deprecated.".\n'
```

File: demisto_sdk/commands/common/errors.py [hunk 7]
```diff
@@ -367,12 +370,27 @@ def invalid_v2_integration_name():
...
     @staticmethod
     @error_code_decorator
     def invalid_v2_script_name():
         return "The name of this v2 script is incorrect , should be **name**V2." \
                " e.g: DBotTrainTextClassifierV2"
 
+    @staticmethod
+    @error_code_decorator
+    def invalid_deprecated_script():
+        return 'The script comment has to start with "Deprecated."'
```
Comment (guykeller) [inline demisto_sdk/commands/common/errors.py:392]: Same here - let's make these changes everywhere
```suggestion
        return 'Every deprecated script's comment has to start with "Deprecated."'
```

File: demisto_sdk/commands/common/hook_validations/playbook.py [hunk 8]
```diff
@@ -241,3 +242,14 @@ def is_root_connected_to_all_tasks(self):  # type: () -> bool
                 return False
 
         return tasks_bucket.issubset(next_tasks_bucket)
+
+    def is_valid_deprecated_playbook(self) -> bool:
+        is_valid = True
+        hidden = self.current_file.get('hidden', False)
```
Comment (guykeller) [inline demisto_sdk/commands/common/hook_validations/playbook.py:248]: change to `is_hidden`, and change in the `script.py` file to `is_deprecated` please

File: demisto_sdk/commands/common/errors.py [hunk 9]
```diff
@@ -367,12 +370,28 @@ def invalid_v2_integration_name():
     def found_hidden_param(parameter_name):
         return f"Parameter: \"{parameter_name}\" can't be hidden. Please remove this field."
 
+    @staticmethod
+    @error_code_decorator
+    def invalid_deprecated_integration(display_name, description):
+        error_answer = ''
```
Comment (guykeller) [inline demisto_sdk/commands/common/errors.py:376]:
```suggestion
        error_message = ''
```

File: demisto_sdk/commands/common/errors.py [hunk 10]
```diff
@@ -367,12 +370,28 @@ def invalid_v2_integration_name():
     def found_hidden_param(parameter_name):
         return f"Parameter: \"{parameter_name}\" can't be hidden. Please remove this field."
 
+    @staticmethod
+    @error_code_decorator
+    def invalid_deprecated_integration(display_name, description):
+        error_message = ''
+        if not display_name.startswith('Deprecated.'):
```
Comment (ronykoz) [inline demisto_sdk/commands/common/errors.py:377]: the test should not be here. it should be in the validator.
we should check for `(Deprecate)` at the end.

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 11]
```diff
@@ -730,6 +731,18 @@ def is_valid_hidden_params(self) -> bool:
 
         return ans
 
+    def is_valid_deprecated_integration(self) -> bool:
+        is_valid = True
+        is_deprecated = self.current_file.get('deprecated', False)
+        display_name = self.current_file.get('display', '')
+        description = self.current_file.get('description', '')
+        if is_deprecated:
+            if not display_name.endswith('(Deprecated)') or not description.startswith('Deprecated.'):
```
Comment (ronykoz) [inline demisto_sdk/commands/common/hook_validations/integration.py:740]: this should be 2 checks

**Review by ronykoz — CHANGES_REQUESTED** [review]

**Review by ronykoz — APPROVED** [review]
